File: Clase/C++/Moto.cc

# include "Vehiculo.cc"

class Moto : public Vehiculo {

    // Atributos de la Clase Moto

    private:
        string motor;

    // Metodos de la Clase Moto

    public:
        Moto(string, string, string, string);
        void setMotor(string m);
        string getMotor();
        void mostrarDatos();
        void tocarClaxon();

};

// Constructor de Clase Moto

Moto::Moto(string mat, string mar, string mod, string mot) : Vehiculo(mat, mar, mod) {

    setMotor(mot);

};

// Setters y Getters

void Moto::setMotor(string m) { this->motor = m; };

string Moto::getMotor() { return this->motor; };

// Metodo para imprimir datos de la Moto

void Moto::mostrarDatos() {

    Vehiculo::mostrarDatos();
    cout << "Motor: " << getMotor() << endl;

};

// Metodo para simular 'tocar el claxon' en la Moto

void Moto::tocarClaxon() {

    cout << "Pam! Pam!" << endl;

};

File: Clase/Java/Moto.java

package clases;

public class Moto extends Vehiculo {
	
	// Atributos de la Clase Moto
	
	String motor;
	
	// Constructor de Clase Moto
	
	public Moto(String mat, String mar, String mod, String mot) {
		
		super(mat, mar, mod);
		setMotor(mot);
		
	}
	
	// Setters y Getters
	
	public void setMotor(String m) { this.motor = m; }
	
	public String getMotor() { return this.motor; }
	
	// Metodo para imprimir datos de la Moto
	
	@Override
	public void mostrarDatos() {
		
		System.out.println("Datos del Vehiculo:");
		System.out.println("Matricula: " + getMatricula());
		System.out.println("Marca: " + getMarca());
		System.out.println("Modelo: " + getModelo());
		System.out.println("Motor: " + getMotor());
		
	}
	
	// Metodo para simular 'tocar el claxon' en la Moto
	
	@Override
	public void tocarClaxon() {
		
		System.out.println("Pam! Pam!");
		
	}
	
}
